helm/gateway-api-config/templates/dnsendpoints.yaml

```diff
@@ -1,6 +1,12 @@
 {{- range $i, $gateway := .Values.gateways }}
-{{- if $gateway.enabled }}
-{{- if and $gateway.dnsEndpoints.enabled $gateway.subdomains }}
+{{- if and $gateway.enabled $gateway.dnsEndpoints.enabled }}
+{{- $subdomains := list }}
+{{- range $j, $listener := $gateway.listeners }}
```

Comment:
I'd say we use a different DNSEndpoint for each listener and move the value inside the listener block.
We might want to disable the DNSEndpoint creation for some particular listener, and in some cases we might need to annotate the resource differently because it might need to be managed by another instance of external-dns